elastalert/elastalert.py

```diff
@@ -978,6 +982,33 @@ def get_top_counts(self, rule, starttime, endtime, keys, number=5, qk=None):
             all_counts['top_events_%s' % (key)] = dict(counts[:number])
         return all_counts
 
+    def next_alert_time(self, rule, name, timestamp):
+        """ Calculate an 'until' time and exponent based on how much past the last 'until' we are. """
+        if name in self.silence_cache:
+            last_until, exponent = self.silence_cache[name]
+        else:
+            # If for some reason this isn't cached, pretend it was a week ago
+            last_until = timestamp - datetime.timedelta(weeks=1)
+            exponent = 0
+
+        if not rule.get('exponential_realert'):
+            return timestamp + rule['realert'], 0
+
+        diff = seconds(timestamp - last_until)
+        # If it's been less than twice the alert time, increase exponent
```

Comment:
In general, I prefer comments that explain why instead of how. Perhaps prose explaining why this works would be more effective here? Otherwise, it's better to remove these comments, as they're not doing anything we can't already read here.